make.py

```diff
@@ -78,52 +92,125 @@ def make_github_issue_no_notify(title, body, labels):
-    "tplink",
-    "tradfri",
-    "unifi_direct",
-    "upnp",
-    "vera",
-    "wunderground",
-    "yr",
-    "zha",
-    "zwave",
-]:
-    title = f"Fix {domain} tests that have uncaught exceptions"
-    body = template.replace("{{ DOMAIN }}", domain)
-    labels = [f"integration: {domain}", "to do"]
-    # print(title)
-    # print(body)
-    # print(labels)
-    make_github_issue(title=title, body=body, labels=labels)
+def common_issue_options(func):
+    """Supply common issue options."""
+    func = click.option(
+        "-t",
+        "--token",
+        prompt=True,
+        hide_input=True,
+        help="Set the auth token.",
+    )(func)
+    func = click.option(
+        "-R",
+        "--repo",
+        default=REPO_NAME,
+        show_default=True,
+        help="Set the target repo.",
+    )(func)
+    func = click.option(
+        "-u",
+        "--username",
+        required=True,
+        help="Set the username.",
+    )(func)
+    func = click.option(
+        "-O",
+        "--owner",
+        default=REPO_OWNER,
+        show_default=True,
+        help="Set the repository owner.",
+    )(func)
+    func = click.option(
+        "-T",
+        "--title",
+        required=True,
+        help="Set the issue title.",
+    )(func)
+    func = click.option(
+        "-b",
+        "--body",
+        type=click.Path(exists=True, dir_okay=False),
+        help="Set file path to a markdown file with issue body.",
+    )(func)
+    func = click.option(
+        "-a",
+        "--assignee",
+        show_default=True,
+        help="Set the issue assignee.",
+    )(func)
+    func = click.option(
+        "-m",
+        "--milestone",
+        help="Set the issue milestone.",
+    )(func)
+    func = click.option(
+        "-d",
+        "--domains",
+        is_flag=True,
+        help="Create one issue per domain.",
+    )(func)
+    func = click.option("-l", "--labels", multiple=True, help="Set the issue labels.")(
+        func
+    )
+    return func
+
+
+@click.group(
+    options_metavar="", subcommand_metavar="<command>", context_settings=SETTINGS
+)
+def cli():
+    """Batch create github.com issues."""
+
+
+@click.command(options_metavar="<options>")
+@click.option(
+    "-s", "--silent", is_flag=True, help="Make an issue without notifications."
+)
+@common_issue_options
+def issue(silent, owner, repo, token, username, title, body, labels, domains, **kwargs):
+    """Create issue on github.com."""
+    token = token or Path(".token").read_text().strip()
+    if body:
+        body = Path(body).read_text()
+
+    # Authentication for user filing issue (must have read/write access to
+    # repository to add issue to)
+    auth = Auth(repo_name=repo, repo_owner=owner, username=username, token=token)
+
+    issue_func = partial(make_github_issue, auth, **kwargs)
+
+    if silent:
+        # the import API doesn't accept optional values
+        issue_func = partial(make_github_issue_no_notify, auth)
+
+    if domains:
```

Comment:
If it's not domains, are you just creating 1 issue ? What would be the use case for that ?

Reply:
Yeah, there should be some batch case otherwise this script isn't useful. I guess we should allow another batch mode than domain to make it more general. But for now we can default to domain.

Reply:
Updated.